cpp/src/shuffler/shuffler.cpp

```diff
@@ -200,10 +200,10 @@ class Shuffler::Progress {
         while (true) {
             auto const [msg, src] = shuffler_.comm_->recv_any(metadata_tag);
             if (msg) {
-                auto chunk = Chunk::from_metadata_message(msg);
+                auto chunk = Chunk::deserialize(*msg, false);
                 log.trace("recv_any from ", src, ": ", chunk);
                 RAPIDSMPF_EXPECTS(
-                    shuffler_.partition_owner(shuffler_.comm_, chunk.pid)
+                    shuffler_.partition_owner(shuffler_.comm_, chunk.part_id(0))
```

Comment:
I guess we only handle singleton chunks here right?
It would be good with TODO and maybe a constexpr instead of `0`.

Reply:
I added a comment here. If the `outgoing_postbox_` guarantees that when it accepts a chunk, all partitions IDs in the chunk maps to the same key, we can safely use the first partition ID.

Reply:
But in the future when we receive a `ChunkBatch`, it might contain different partitions IDs right? So we could use a TODO saying that we will need some code to split `ChunkBatch` based on their partition IDs here?

Reply:
I see what you mean. Let me add some todos

Reply:
@madsbk but coming to think of it, at this point, if the `outgoing_postbox_` gurantees that it accepts only `Chunk`s that has partition IDs map to the same rank, then we only have to check the first part ID here, isnt it?

Reply:
That is true. I guess we need the TODO where we insert into `outgoing_postbox_`